### PR TITLE
Add timeout option for kubernets

### PR DESCRIPTION
**Pull request by adrianlzt**
Patch provided by jchauncey

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [ ] Associated README.md updated.
- [ ] Has appropriate unit tests.

### CODE REVIEW

**Comment by adrianlzt** [comment]
This code is provided by @jchauncey in #3203

File: plugins/inputs/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -101,10 +107,17 @@ func (k *Kubernetes) gatherSummary(baseURL string, acc telegraf.Accumulator) err
 	}
 
 	if k.RoundTripper == nil {
+		// Set default values
+		if k.ResponseTimeout.Duration < time.Second {
+			k.ResponseTimeout.Duration = time.Second * 5
+		}
+		if err != nil {
```
Comment (danielnelson) [inline plugins/inputs/kubernetes/kubernetes.go]: This err check should go away, there is nothing to check here.

File: plugins/inputs/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -101,10 +107,17 @@ func (k *Kubernetes) gatherSummary(baseURL string, acc telegraf.Accumulator) err
...
+			return err
+		}
 		k.RoundTripper = &http.Transport{
 			TLSHandshakeTimeout:   5 * time.Second,
 			TLSClientConfig:       tlsCfg,
-			ResponseHeaderTimeout: time.Duration(3 * time.Second),
+			ResponseHeaderTimeout: k.ResponseTimeout.Duration,
```
Comment (danielnelson) [inline plugins/inputs/kubernetes/kubernetes.go]: Just thinking aloud, we should switch to a http.client sometime so we can include the time to get the body.  We are no worse off now with this change so not required.

**Comment by danielnelson** [comment]
@adrianlzt I can fix the above issues if that's easier.

**Comment by adrianlzt** [comment]
sure @danielnelson

**Pull request merged**